embedded/sql/engine_test.go

```diff
@@ -386,6 +386,55 @@ func TestTimestampCasts(t *testing.T) {
 
 }
 
+func TestNowFunctionEvalsToTxTimestamp(t *testing.T) {
+	defer os.RemoveAll("tx_timestamp")
+
+	st, err := store.Open("tx_timestamp", store.DefaultOptions())
+	require.NoError(t, err)
+	defer closeStore(t, st)
+
+	engine, err := NewEngine(st, DefaultOptions().WithPrefix(sqlPrefix))
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("CREATE DATABASE db1", nil, nil)
+	require.NoError(t, err)
+
+	err = engine.SetCurrentDatabase("db1")
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("CREATE TABLE tx_timestamp (id INTEGER AUTO_INCREMENT, ts TIMESTAMP, PRIMARY KEY id)", nil, nil)
+	require.NoError(t, err)
+
+	currentTs := time.Now()
+
+	for it := 0; it < 3; it++ {
+		tx, _, err := engine.Exec("BEGIN TRANSACTION;", nil, nil)
+		require.NoError(t, err)
+
+		require.True(t, tx.Timestamp().After(currentTs))
+
+		for i := 0; i < 5; i++ {
+			_, _, err = engine.Exec("INSERT INTO tx_timestamp(ts) VALUES (NOW()), (NOW())", nil, tx)
+			require.NoError(t, err)
+		}
+
+		_, _, err = engine.Exec("COMMIT;", nil, tx)
+		require.NoError(t, err)
+
+		r, err := engine.Query("SELECT * FROM tx_timestamp WHERE ts = @ts", map[string]interface{}{"ts": tx.Timestamp()}, nil)
+		require.NoError(t, err)
+		defer r.Close()
+
+		for i := 0; i < 10; i++ {
+			row, err := r.Read()
+			require.NoError(t, err)
+			require.EqualValues(t, tx.Timestamp().UTC(), row.ValuesBySelector[EncodeSelector("", "db1", "tx_timestamp", "ts")].Value())
+		}
+
```

Comment:
Please check here if attempt to read one more results in ErrNoMoreRows

Reply:
not completely sure about that, if tx processing is fast enough, there could be more than one transaction "at the same time".. lets discuss it

Reply:
True, more Txs can end up with the same timestamp. But how then would we make sure that the tx timestamp is changing?
I think we can use `time.Sleep` here since it has the lower-bound of delay guaranteed (like sleep 2 microseconds between rounds).